pingcap/tidb
# store : add store limit to restrain bad store from occupying too much token limit.

### What problem does this PR solve? 
In some cases, when a store is broken in tikv cluster, the store will hang all write requests sent to it, which will cause token limit exhausted in tidb server, consequently leading QPS to decrease sharply.



### What is changed and how it works?
Add storeLimit in `Store` structure, we should do the count on it before sending the request to each store, eventually, control the traffic of each store. 

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Store : add store limit check before sending request

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

Release note

 - add store limit to restrain bad store from occupying too much token limit.
